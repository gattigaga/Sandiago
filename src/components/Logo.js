import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Text = styled.Text`
  color: white;
  font-family: Lobster-Regular;
  font-size: 48px;
`;

const Logo = ({ style }) => <Text style={style}>Sandiago</Text>;

Logo.propTypes = {
  style: PropTypes.array
};

export default Logo;
