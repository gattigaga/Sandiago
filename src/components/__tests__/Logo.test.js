import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Logo from "../Logo";

describe("Logo", () => {
  const setup = propOverrides => {
    const props = {
      ...propOverrides
    };

    const wrapper = shallow(<Logo {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with custom style", () => {
    const { wrapper } = setup({ style: [{ color: "red" }] });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
