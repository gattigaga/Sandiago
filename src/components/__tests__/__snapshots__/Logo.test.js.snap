// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should renders default 1`] = `
<Styled(Text)>
  Sandiago
</Styled(Text)>
`;

exports[`Logo should renders with custom style 1`] = `
<Styled(Text)
  style={
    Array [
      Object {
        "color": "red",
      },
    ]
  }
>
  Sandiago
</Styled(Text)>
`;
